src/CartBlock.C

```diff
@@ -80,35 +79,22 @@ void CartBlock::getInterpolatedData(int *nints,int *nreals,int **intData,
     {
       nintold=(*nints);
       nrealold=(*nreals);
-      if (nintold > 0) {
-       tmpint=(int *)malloc(sizeof(int)*3*(*nints));
-       tmpreal=(double *)malloc(sizeof(double)*(*nreals));
-       for(i=0;i<(*nints)*3;i++) tmpint[i]=(*intData)[i];
-       for(i=0;i<(*nreals);i++) tmpreal[i]=(*realData)[i];
-       //
-       TIOGA_FREE((*intData));
-       TIOGA_FREE((*realData)); // didnt free this before ??
-       //  
-      }
+
+      // resize interpolation data vectors
       (*nints)+=interpCount;
       (*nreals)+=(interpCount*(nvar_cell+nvar_node));
-      (*intData)=(int *)malloc(sizeof(int)*3*(*nints));
-      (*realData)=(double *)malloc(sizeof(double)*(*nreals));
-      if (nintold > 0) {
-       for(i=0;i<nintold*3;i++) (*intData)[i]=tmpint[i];
-       for(i=0;i<nrealold;i++) (*realData)[i]=tmpreal[i];      
-       TIOGA_FREE(tmpint);
-       TIOGA_FREE(tmpreal);
-      }
+      intData.resize(3*(*nints));
+      realData.resize(*nreals);
+
       listptr=interpList;
-      icount=3*nintold;
+      icount=3*nintold; icount=3*nintold;
```

Comment:
statement repeated here ?

Reply:
@jsitaraman Fixed it in `c0718b093c605da25d916f1c9ccb5d8e1d8203aa`